src/VecSim/algorithms/brute_force/bfs_batch_iterator.h

```diff
@@ -6,7 +6,7 @@
 
 class BFS_BatchIterator : public BF_BatchIterator {
 public:
-    BFS_BatchIterator(void *query_vector, const BruteForceIndex *index,
+    BFS_BatchIterator(void *query_vector, const BruteForceIndex<float, float> *index,
```

Comment:
This causes some compile errors and is not really necessary because there should not be any difference between the classes besides implementation. I might change it after we add the batch iterator factory.